iml-agent/src/action_plugins/check_stonith.rs

```diff
@@ -0,0 +1,314 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::{
+    agent_error::{CibError, ImlAgentError},
+    cmd::cmd_output,
+};
+use elementtree::Element;
+use futures::Future;
+use iml_wire_types::{ComponentState, ConfigState, ServiceState, RunState};
+use std::default::Default;
+use std::process::Command;
+use tracing::{span, Level};
+use tracing_futures::Instrument;
+
+fn inst_attr_problem(elem: &Option<&Element>, node: &String) -> Option<String> {
+    match elem {
+        None => None,
+        Some(instance) => {
+            let mut hc: Option<bool> = None;
```

Comment:
This feels really busy. Is there any way this could be cleaned up? We might be able to pull some of the logic into functions. What are your thoughts?